sqllogictest/src/runner.rs

```diff
@@ -328,6 +342,135 @@ impl<D: AsyncDB> Runner<D> {
+/// db address, we can use conn_builder to connect multiple target(in prepare).
+///
+/// ```no_run
+/// let db = DB::connect(hosts, dbname).await;
+/// let mut tester = sqllogictest::ParallelRunner::new(db, hosts.to_vec());
+/// let tasks = tester
+///     .prepare_async(glob, DB::connect)
+///     .await
+///     .expect("prepare failed");
+/// tester
+///     .run_parallel_async(tasks, jobs)
+///     .await
+///     .expect("test failed");
+/// ```
+
+pub struct ParallelRunner<D: AsyncDB> {
+    db: D,
+    hosts: Vec<String>,
+}
+
+impl<D: AsyncDB + 'static> ParallelRunner<D> {
+    pub fn new(db: D, hosts: Vec<String>) -> ParallelRunner<D> {
+        ParallelRunner { db, hosts }
+    }
+
+    /// parse slt files and create DB connection for multiple target. Here we use the conn_builder
+    /// parameter function for connection creating, the function accept two parameter which is host
+    /// and dbname in order.
+    pub async fn prepare_async<F, Fut>(
+        &mut self,
+        glob: &str,
+        conn_builder: F,
+    ) -> Result<Vec<(D, String)>, TestError>
+    where
+        F: Fn(String, String) -> Fut,
+        Fut: Future<Output = D>,
+    {
+        let files = glob::glob(glob).expect("failed to read glob pattern");
+        let mut tasks = vec![];
+        let mut idx = 0;
+
+        for file in files {
+            // for every slt file, we create a database against table conflict
+            let file = file.unwrap();
+            let db_name = file
+                .file_name()
+                .expect("not a valid filename")
+                .to_str()
+                .expect("not a UTF-8 filename");
+            let db_name = db_name
+                .replace(' ', "_")
+                .replace('.', "_")
+                .replace('-', "_");
+
+            self.db
+                .run(&format!("CREATE DATABASE {};", db_name))
+                .await
+                .expect("create db failed");
+            tasks.push((
+                conn_builder(self.hosts[idx % self.hosts.len()].clone(), db_name).await,
+                file.to_string_lossy().to_string(),
+            ));
+            idx += 1;
+        }
+
+        Ok(tasks)
+    }
+
+    /// sync version of `prepare_aync`
+    pub fn prepare<F, Fut>(
+        &mut self,
+        glob: &str,
+        conn_builder: F,
+    ) -> Result<Vec<(D, String)>, TestError>
+    where
+        F: Fn(String, String) -> D,
+    {
+        future::block_on(
+            self.prepare_async(glob, |host, dbname| async { conn_builder(host, dbname) }),
+        )
+    }
+
+    /// aceept the tasks, spawn jobs task to run slt test. the tasks are (AsyncDB, slt filename)
+    /// pairs.
+    pub async fn run_parallel_async(
+        &self,
+        mut tasks: Vec<(D, String)>,
+        jobs: usize,
+    ) -> Result<(), TestError> {
+        let chunk_size = (tasks.len() as f64 / jobs as f64).ceil() as usize;
```

Comment:
prefer integer operations to floating point
```suggestion
        let chunk_size = (tasks.len() + jobs - 1) / jobs;
```